### PR TITLE
IDE-641 DOCS:Other language functionality

### PR DESCRIPTION
**Pull request by g-pan**
Fix IDE-641 DOCS:Other language functionality
Document the ECL IDE functionality
for other languages in addition to ECL.

@JamesDeFabia please review
@dehilsterlexis please review 
PDF and Build Test output can be obtained from http://10.240.32.243/view/Docs/job/DocDevII/
(this content can be found on pg 40-41 of the ECL IDE and Client Tools doc)

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Successful Build Test http://10.240.32.243/view/Docs/job/DocDevII/

### CODE REVIEW

File: docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml [hunk 1]
```diff
@@ -3185,6 +3177,155 @@ ENDMACRO;
...
+                    <entry><emphasis
+                    role="bold">Functionality</emphasis></entry>
+                  </row>
+
+                  <row>
+                    <entry>ECL</entry>
+
+                    <entry><itemizedlist>
+                        <listitem>
+                          <para>Colorized file differentiation</para>
```
Comment (JamesDeFabia) [inline docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml]: Colorized file differentiation is confusing. What you are trying to say is: 
Repository icon is colorized to identify the file type.

File: docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml [hunk 2]
```diff
@@ -3185,6 +3177,155 @@ ENDMACRO;
...
+                        </listitem>
+
+                        <listitem>
+                          <para>Syntax Check</para>
+                        </listitem>
+
+                        <listitem>
+                          <para>Submit</para>
+                        </listitem>
+
```
Comment (JamesDeFabia) [inline docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml]: also Submit Selected

File: docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml [hunk 3]
```diff
@@ -3185,6 +3177,155 @@ ENDMACRO;
...
+                    <entry>ECL</entry>
+
+                    <entry><itemizedlist>
+                        <listitem>
+                          <para>Colorized file differentiation</para>
+                        </listitem>
+
+                        <listitem>
+                          <para>Colorized keywords</para>
+                        </listitem>
```
Comment (JamesDeFabia) [inline docs/HPCCClientTools/CT_Mods/CT_ECL_IDE.xml]: Syntax Highlighting

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Comments inline. Be aware that the same changes need to be made for each file type.

**Comment by g-pan** [comment]
Updated. Added changes based on comments. @JamesDeFabia please review
@dehilsterlexis please review

**Review by JamesDeFabia — APPROVED** [review]
Looks good to merge

**Comment by dehilsterlexis** [comment]
Looks good to me

**Pull request merged**